README.md

```diff
@@ -6,10 +6,96 @@ The package provide a widget called `PaginatedLiveList` that takes a concrete `P
 
 For usage example see the example directory.
 
+## Installation
+
+Add `paginated_live_list` as a dependency to your project `pubspec.yaml` and run `flutter pub get` to install all your dependencies.
+
+```yaml
+# pubspec.yaml
+dependencies:
+  paginated_live_list:
+```
+
+## Basic Concepts
+
+### `PageCursor` class:
+
+serve as a pointer to the next page. Every page has its own cursor value. if it was null means we want to point the first page (since it has not pereceding pages). the developer can adjust how he want to implement the cursor. this a user's define value. see the example below.
+
+```dart
+class IntPageCursor implements PageCursor {
+  final int next;
+
+  IntPageCursor(this.next);
+}
+```
+
+### `Page` Generic class
+contains the list `items` of the page and a `cursor` defined by the user. also it has a boolean indentifier to indentify if this is the last page or not.
+
+## Import the package
+
+
+```dart
+import 'package:paginated_live_list/paginated_live_list.dart';
+```
+
 ## Implementing a PaginationController
 
-You need to implement `Page` comparison with `PaginationController.compareTo`, `PageCursor` adjustment with `PaginationController.adjustCursor`, and loading of new pages with `PaginationController.onLoadPage`.
+ You need to implement `Page` comparison with `PaginationController.compareTo`, `PageCursor` adjustment with `PaginationController.adjustCursor`, and loading of new pages with `PaginationController.onLoadPage`.
 
 `PageCursor` adjustment is needed when a page changes it's cursor due to an update. The adjustment allows the PaginationController to maintain the current pages by not having to reload all the pages after the page that changed it's cursor.
 
+### random_controller.dart
+```dart
+class RandomController extends PaginationController<Offer> {
+    /// YOUR CODE GOES HERE
+}
+```
+
+### Defining your own cursors
+
+### random_page_cursor.dart
+```dart
+class RandomPageCursor implements PageCursor {
+    final int randomCursor;
+
+    RandomPageCursor(this.randomCursor);
+}
+```
+
+then you can use this in your custom controller or where ever you create your cursors.
+
+```dart
+Page<T> createPageCursor(Page<T> page) {
+    final last = page.items.last;
+    int randomNumber = random.nextInt(100)
+    return Page(
+        page.items,
+        RandomPageCursor(DateTime.now().millisecondsSinceEpoch),
+        page.isLastPage);
+  }
```

Comment:
remove `randomNumber`, and make the example more realistic (a simplified implmentation of `onLoadPage`)

```suggestion
Stream<Page<T>> onLoadPage(IndexedPageCursor cursor) {
    return dataSource.list(cursor.page, size: 10).map((items) => Page(
      items,
      IndexedPageCursor(cursor.page + 1),
      items.length < 10,
    ));
  }
```